saleor/static/dashboard-next/components/ColumnPicker/ColumnPicker.tsx

```diff
@@ -0,0 +1,103 @@
+import ClickAwayListener from "@material-ui/core/ClickAwayListener";
+import Grow from "@material-ui/core/Grow";
+import Popper from "@material-ui/core/Popper";
+import { Theme } from "@material-ui/core/styles";
+import { fade } from "@material-ui/core/styles/colorManipulator";
+import makeStyles from "@material-ui/styles/makeStyles";
+import React from "react";
+
+import ColumnPickerButton from "./ColumnPickerButton";
+import ColumnPickerContent, {
+  ColumnPickerContentProps
+} from "./ColumnPickerContent";
+
+export interface ColumnPickerProps extends ColumnPickerContentProps {
+  className?: string;
+  initial?: boolean;
+}
+
+const useStyles = makeStyles(
+  (theme: Theme) => ({
+    popper: {
+      boxShadow: `0px 5px 10px 0 ${fade(theme.palette.common.black, 0.05)}`,
+      marginTop: theme.spacing.unit * 2,
+      zIndex: 1
+    }
+  }),
+  {
+    name: "ColumnPicker"
+  }
+);
+
+const ColumnPicker: React.FC<ColumnPickerProps> = props => {
+  const {
+    className,
+    columns,
+    initial,
+    selectedColumns,
+    onCancel,
+    onColumnToggle,
+    onReset,
+    onSave
+  } = props;
+  const classes = useStyles(props);
+  const anchor = React.useRef<HTMLDivElement>();
+  const [isExpanded, setExpansionState] = React.useState(false);
+
+  React.useEffect(() => {
+    setTimeout(() => setExpansionState(!!initial), 100);
+  }, []);
+
+  const handleCancel = React.useCallback(() => {
```

Comment:
this function will only fire the initial `onCancel` provided, this may be how you want it to work - so just making sure it's intended